### PR TITLE
Add ability to upload images to conversations

### PR DESCRIPTION
**Pull request by sternetj**
## Changes

  - Users can add up to 10 images at a time and include text with their message
  - Each image is displayed in its own bubble
  - Made the image uploading library an optional dependency since it might not be used by all consumers

## Screenshots

https://github.com/lifeomic/react-native-sdk/assets/2295908/d2bc6299-de64-4937-9df4-511f138c4c77

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: 14
+          node-version: 18
```
Comment (rob-snyder) [inline .github/workflows/release.yml:19]: Man we still have however many repos on node 16 or earlier. Waiting  for AWS to just pull that rug out from under us.

File: src/hooks/Circles/usePrivatePosts.ts [hunk 2]
```diff
@@ -268,3 +321,131 @@ const createPrivatePostMutationDocument = gql`
...
+
+    const attachment: Attachment = {
+      externalId: res.privatePostFileUploadUrl.id,
+      subType: 's3',
+      type: {
+        JPEG: AttachmentType.IMAGE,
+        JPG: AttachmentType.IMAGE,
+        PNG: AttachmentType.IMAGE,
+        GIF: AttachmentType.IMAGE,
+      }[fileType],
```
Comment (rob-snyder) [inline src/hooks/Circles/usePrivatePosts.ts:416]: Feel like I'm missing something, couldn't this just be

```suggestion
      type: AttachmentType.IMAGE,
```

**Review by rob-snyder — DISMISSED** [review]
This seems good, though I have little familiarity with the react-native-sdk. Nice job working through the unit tests. It's a PITA when you can't reproduce locally

File: src/hooks/Circles/usePrivatePosts.ts (hunk 2 above)
Comment (sternetj) [reply src/hooks/Circles/usePrivatePosts.ts:416]: You're not missing anything. This is a relic of an earlier iteration. I originally was trying to add videos too but it had some issues and I wanted to move on and nobody was asking for video uploads.

**Review by rob-snyder — DISMISSED** [review]

File: src/screens/DirectMessagesScreen.tsx [hunk 3]
```diff
@@ -177,32 +291,152 @@ export const DirectMessagesScreen = ({
...
+            style={[
+              { height: styles.pendingImagesContainer?.height },
+              styles.footer,
+            ]}
+          />
+        )
+      }
+      {...({
+        // accessoryStyle overrides are not documented but are supported
+        accessoryStyle: {
```
Comment (bruce-glazier) [inline src/screens/DirectMessagesScreen.tsx:345]: That's fun.

**Review by rob-snyder — APPROVED** [review]

**Review by bruce-glazier — APPROVED** [review]
This turned out to be quite the task. Everything looks good to me!

**Comment by bruce-glazier** [comment]
Only NIT: I feel like the photos would look better without being within a message bubble.

**Pull request merged**